Update Soroban Getting Started (#2)
AhaLabs/soroban-docs: docs/getting-started/deploy-incrementor-to-testnet.mdx

---
sidebar_position: 5
title: 4. Deploy Incrementor to Testnet
description: Deploy the Incrementor contract to Testnet.
---

<head>

>   <title>Write a smart contract that stores and retrieves data.</title>

Reviewer: ```suggestion
  <title>4. Deploy Incrementor to Testnet.</title>
```

Do we need this at all, though? I haven't seen the `<head>` tag on other pages. Nice feature, when it's needed, but it seems like maybe we don't need to customize the defaults on this page.

Reviewer: Oh, it looks like it is on the other pages now. Huh.

Author: I agree that maybe it's not 100% necessary, but just added it to be consistent with the other pages. I'm happy to rethink/remove this as we continue to refine these pages.

  <meta charSet="utf-8" />
  <meta
    property="og:title"
    content="Deploy the Incrementor contract to Testnet."
  />
  <meta
    property="og:description"
    content="Deploy the Incrementor contract to Testnet."
  />
  <link
    rel="canonical"

>     href="https://soroban.stellar.org/docs/getting-started/deployt-incrementor-to-testnet"

Reviewer: ```suggestion
    href="https://soroban.stellar.org/docs/getting-started/deploy-incrementor-to-testnet"
```

  --network testnet \
  --source alice \
  --wasm target/wasm32-unknown-unknown/release/incrementor.wasm
```

This returns the hash of the Wasm bytes. Now you can use `--wasm-hash` with `deploy` rather than `--wasm`. Let's also automatically pipe the returned contract ID into a file in the `.soroban` directory, so that when you search your command history and reuse the deploy command in the future, you don't forget that step (you might want to go back and do something similar with the Hello World deploy, too):

```bash
soroban contract deploy \
  --wasm-hash [paste the output from the last command] \
  --source alice \
  --network testnet \
  > .soroban/incrementor-id
```

You can check that it saved the contract ID correctly:

```bash
cat .soroban/incrementor-id
```

Now you can interact with it over RPC like you did with the Hello World contract:

```bash
soroban contract invoke \
  --id $(cat .soroban/incrementor-id) \
  --source alice \
  --network testnet \
  -- \
  increment
```

You should see the following output:

```bash
1
```

Run it a few more times to watch the count change.

## Run your own network/node

Sometimes you'll need to run your own node:

> - Production apps! It's a bad idea to rely on the public, Stellar-maintained Soroban RPC endpoint. Instead, you should run your own node, and point your app at that.

Reviewer: I've learned more since writing this copy (I think this was copy-pasted from what I wrote?)

```suggestion
- Production apps! Stellar maintains public test RPC nodes for Testnet and Futurenet, but not for Mainnet. Instead, you will need to run your own node, and point your app at that. If you want to use a software-as-a-service platform for this, [various providers](../reference/rpc-list.mdx) are available.
```

Author: Yep, this was copy-pasted from the existing Deploy to Testnet [page](https://soroban.stellar.org/docs/getting-started/deploy-to-testnet#run-your-own-networknode) - this is a great update! 👍

- When you need a network that differs from the version deployed to Testnet.

The Soroban team maintains Docker containers that makes this as straightforward as possible. See the [RPC](../reference/rpc.mdx) reference for details.

> :::info It's also possible to run a contract on a fully featured local network. See [the RPC Reference page](../reference/rpc.mdx) for more details. :::
> 

Reviewer: This info box basically just restates the line above

```suggestion
```
